academic-observatory-workflows/academic_observatory_workflows/crossref_metadata_telescope/telescope.py

```diff
@@ -0,0 +1,326 @@
+    :param gke_volume_path: Where to mount the persistent volume
+    :param gke_zone: The zone containing the gke cluster
+    :param gke_volume_size: The amount of storage to request for the persistent volume in GiB
+    :param gke_resource_overrides: Task resource overrides
+    """
+
+    dag_id: str
+    cloud_workspace: CloudWorkspace
+    bq_dataset_id: str = "crossref_metadata"
+    bq_table_name: str = "crossref_metadata"
+    api_bq_dataset_id: str = "crossref_metadata"
+    schema_folder: str = project_path("crossref_metadata_telescope", "schema")
+    dataset_description: str = (
+        "The Crossref Metadata Plus dataset: https://www.crossref.org/services/metadata-retrieval/metadata-plus/"
+    )
+    table_description: str = (
+        "The Crossref Metadata Plus dataset: https://www.crossref.org/services/metadata-retrieval/metadata-plus/"
+    )
+    crossref_metadata_conn_id: str = "crossref_metadata"
+    max_processes: int = os.cpu_count()
+    batch_size: int = 20
+    start_date: pendulum.DateTime = pendulum.datetime(2020, 6, 7)
+    schedule: str = "0 0 7 * *"
+    catchup: bool = True
+    queue: str = "default"
+    max_active_runs: int = 1
+    retries: int = 3
+    gke_image = "us-docker.pkg.dev/academic-observatory/academic-observatory/academic-observatory:latest"
+    gke_namespace: str = "coki-astro"
+    gke_startup_timeout_seconds: int = 300
+    gke_conn_id: str = "gke_cluster"
+    docker_astro_uid: int = 50000
+    gke_volume_name: str = "crossref-metadata"
+    gke_volume_path: str = "/data"
+    gke_zone: str = "us-central1"
+    gke_volume_size: int = 2500
+    gke_resource_overrides: Optional[dict] = None
+
+
+def create_dag(dag_params: DagParams) -> DAG:
+    """The Crossref Metadata DAG"""
+
+    task_resources = _TaskResources(overrides=dag_params.gke_resource_overrides)
+    # Common @task.kubernetes params
+    volume_mounts = [k8s.V1VolumeMount(mount_path=dag_params.gke_volume_path, name=dag_params.gke_volume_name)]
+    volumes = [
+        k8s.V1Volume(
+            name=dag_params.gke_volume_name,
+            persistent_volume_claim=k8s.V1PersistentVolumeClaimVolumeSource(claim_name=dag_params.gke_volume_name),
+        )
+    ]
+    kubernetes_task_params = dict(
+        image=dag_params.gke_image,
+        security_context=k8s.V1PodSecurityContext(
+            fs_group=dag_params.docker_astro_uid,
+            fs_group_change_policy="OnRootMismatch",
+            run_as_group=dag_params.docker_astro_uid,
+            run_as_user=dag_params.docker_astro_uid,
+        ),
+        do_xcom_push=True,
+        get_logs=True,
+        in_cluster=False,
+        kubernetes_conn_id=dag_params.gke_conn_id,
+        log_events_on_failure=True,
+        namespace=dag_params.gke_namespace,
+        startup_timeout_seconds=dag_params.gke_startup_timeout_seconds,
+        env_vars={"DATA_PATH": dag_params.gke_volume_path},
+        volumes=volumes,
+        volume_mounts=volume_mounts,
+    )
```

Comment:
I would make this into a reusable function, since it will be used for each workflow. This is also why putting the GKE related params into their own data class might be good, then `dag_params: DagParams` could be changed to that data class specifically.  

```python
def make_kubernetes_task_params(dag_params: DagParams):
    volume_mounts = [k8s.V1VolumeMount(mount_path=dag_params.gke_volume_path, name=dag_params.gke_volume_name)]
    volumes = [
        k8s.V1Volume(
            name=dag_params.gke_volume_name,
            persistent_volume_claim=k8s.V1PersistentVolumeClaimVolumeSource(claim_name=dag_params.gke_volume_name),
        )
    ]
    return dict(
        image=dag_params.gke_image,
        security_context=k8s.V1PodSecurityContext(
            fs_group=dag_params.docker_astro_uid,
            fs_group_change_policy="OnRootMismatch",
            run_as_group=dag_params.docker_astro_uid,
            run_as_user=dag_params.docker_astro_uid,
        ),
        do_xcom_push=True,
        get_logs=True,
        in_cluster=False,
        kubernetes_conn_id=dag_params.gke_conn_id,
        log_events_on_failure=True,
        namespace=dag_params.gke_namespace,
        startup_timeout_seconds=dag_params.gke_startup_timeout_seconds,
        env_vars={"DATA_PATH": dag_params.gke_volume_path},
        volumes=volumes,
        volume_mounts=volume_mounts,
    )
```